src/components/Divider.tsx

```diff
@@ -1,70 +1,141 @@
+  return useSelector(
+    (state: State) => {
+      const parentPath = rootedParentOf(state, path)
+      const parentId = head(parentPath)
+      const grandParentPath = parentId ? rootedParentOf(state, parentPath) : []
+      const grandParentId = head(grandParentPath)
+      const children = parentId ? getAllChildrenAsThoughts(state, parentId) : []
+      const childrenWithoutDividers = children.filter(child => !isDivider(child.value))
+      const isOnlyChild = childrenWithoutDividers.length === 0
+      const isTableView =
+        (parentId && attributeEquals(state, parentId, '=view', 'Table')) ||
+        (grandParentId && attributeEquals(state, grandParentId, '=view', 'Table'))
+
+      let thoughtsAtSameDepth: { id: ThoughtId; value: string }[] = []
+
+      if (isTableView && isOnlyChild && grandParentId) {
+        const parentSiblings = getAllChildrenAsThoughts(state, grandParentId).filter(child => !isDivider(child.value))
+        thoughtsAtSameDepth = parentSiblings.flatMap(parent => {
+          const childrenOfParent = parent.id ? getAllChildrenAsThoughts(state, parent.id) : []
+          return childrenOfParent.filter(child => !isDivider(child.value))
+        })
+      }
+
+      return {
+        isOnlyChild,
+        isTableView,
+        children,
+        thoughtsAtSameDepth,
+      }
+    },
+    // Ensure the selector runs only when values change, excluding ref changes to avoid redundant re-renders.
+    (prev, next) =>
+      prev.isOnlyChild === next.isOnlyChild &&
+      prev.isTableView === next.isTableView &&
+      prev.children.length === next.children.length &&
+      prev.children.every((child, index) => child.value === next.children[index].value) &&
+      prev.thoughtsAtSameDepth.length === next.thoughtsAtSameDepth.length &&
+      prev.thoughtsAtSameDepth.every((thought, index) => thought.value === next.thoughtsAtSameDepth[index].value),
+  )
+}
+
+/**
+ * Calculates the width of multiple thoughts by measuring their rendered widths in the DOM.
+ */
+const getThoughtWidths = (thoughts: { id: ThoughtId }[]): number[] => {
+  return thoughts.map(thought => {
+    const innerThoughtElement = document.querySelector(
+      `[aria-label="tree-node"][data-id="${thought.id}"] [aria-label="thought"]`,
+    )
+    return innerThoughtElement?.getBoundingClientRect().width ?? 0
+  })
+}
 
 /** A custom horizontal rule. */
 const Divider = ({ path, cssRaw }: { path: Path; cssRaw?: SystemStyleObject }) => {
-  const dividerSetWidth = React.createRef<HTMLInputElement>()
   const dispatch = useDispatch()
+  const dividerRef = useRef<HTMLDivElement>(null)
+  const [dividerWidth, setDividerWidth] = useState<number>(DIVIDER_MIN_WIDTH)
+  const { isOnlyChild, isTableView, children, thoughtsAtSameDepth } = useDividerData(path)
+  const editingThoughtId = useSelector((state: State) => state.cursor && head(state.cursor))
+  const editingValueUntrimmed = editingValueUntrimmedStore.useSelector(editingValue => editingValue)
```

Comment:
Shorthand for selecting the whole state: `editingValueUntrimmedStore.useState()`